Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Tooltips/ToolTip.cs

```diff
@@ -0,0 +1,540 @@
+    /// </summary>
+    public class ToolTip : MonoBehaviour//, IMixedRealityInputHandler
+    {
+        [SerializeField]
+        private bool showBackground = true;
+        /// <summary>
+        /// getter/setter for showing the opaque background of tooltip.
+        /// </summary>
+        public bool ShowBackground
+        {
+            get
+            {
+                return showBackground;
+            }
+            set
+            {
+                showBackground = value;
+                GetComponent<ToolTipBackgroundMesh>().IsVisible = value;
+            }
+        }
+
+        [SerializeField]
+        private bool showOutline = false;
+        /// <summary>
+        /// getter/setter for showing white trim around edge of tooltip
+        /// </summary>
+        public bool ShowOutline
+        {
+            get
+            {
+                return showOutline;
+            }
+            set
+            {
+                showOutline = value;
+                GameObject TipBackground = contentParent.transform.GetChild(1).gameObject;
+                RectangleLineDataProvider rectangle = TipBackground.GetComponent<RectangleLineDataProvider>();
+                rectangle.enabled = value;
+            }
+        }
+
+        [SerializeField]
+        private bool showConnector = true;
+        /// <summary>
+        /// getter/setter for showing connecting stem between tooltip and parent object
+        /// </summary>
+        public bool ShowConnector
+        {
+            get
+            {
+                return showConnector;
+            }
+            set
+            {
+                showConnector = value;
+                //todo fix this
+
+                BaseMixedRealityLineDataProvider lineScript = GetComponent<BaseMixedRealityLineDataProvider>();
+                lineScript.enabled = value;
+            }
+        }
+
+        [SerializeField]
+        private DisplayModeType tipState;
+        /// <summary>
+        /// getter/setter for the display state of a tooltip
+        /// </summary>
+        public DisplayModeType TipState
+        {
+            get
+            {
+                return tipState;
+            }
+            set
+            {
+                tipState = value;
+            }
+        }
+
+        [SerializeField]
+        private DisplayModeType groupTipState;
+        /// <summary>
+        /// getter/setter for display state of group of tooltips
+        /// </summary>
+        public DisplayModeType GroupTipState
+        {
+            set
+            {
+                groupTipState = value;
+            }
+            get
+            {
+                return groupTipState;
+            }
+        }
+
+        [SerializeField]
+        private DisplayModeType masterTipState;
+        /// <summary>
+        /// getter/setter for display state of master tooltip
```

Comment:
These comments feel a lot like "gets or sets the foo property" is there more than can be said?

Reply:
Fixed in my PR opened on his branch.